src/python/grpcio/grpc/__init__.py

```diff
@@ -833,7 +833,11 @@ class HandlerCallDetails(six.with_metaclass(abc.ABCMeta)):
 
 
 class GenericRpcHandler(six.with_metaclass(abc.ABCMeta)):
```

Comment:
Even beyond that, I don't understand the issue with having SerivceRpcHandler have the same semantics as GenericHandler.  I think avoiding adding complexity to the API is sufficient to stick with GenericRpcHandler.